tests/dial_nonexisting.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source "$SCRIPT_DIR"/lib/asserts.sh
+
+N1="test_n1"
+N1_SEED=1
+N1_ADDR="/ip6/::1/udp/53137/quic-v1"
+N1_BAD_ADDR="/ip6/::2/udp/53137/quic-v1"
+N1_BAD_PORT="/ip6/::1/udp/53199/quic-v1"
+N2="test_n2"
+N2_SEED=2
+
+echo "Dial nonexisting node test:"
+
+# run daemon
+killall liberum_core &> /dev/null
+cargo run -p liberum_core -- --daemon  &> /dev/null &
+sleep 0.5; # the socket file is created asynchronously and may not be ready yet :))))
+
+# create ndoes
+cargo run -p liberum_cli -- -d new-node $N1 --id-seed $N1_SEED 2> /dev/null
+cargo run -p liberum_cli -- -d new-node $N2 --id-seed $N2_SEED 2> /dev/null
+cargo run -p liberum_cli -- -d config-node $N1 add-external-addr $N1_ADDR 2> /dev/null
+
+# start n1 and get its peer id
+cargo run -p liberum_cli -- -d start-node $N1 2> /dev/null
+N1_ID=$(cargo run -p liberum_cli -- -d get-peer-id $N1 2> /dev/null)
+
+cargo run -p liberum_cli -- -d start-node $N2 2> /dev/null
+
+init_asserts
+
+# dial nonexisting addresses
+RESULT1=$(cargo run -p liberum_cli -- -d dial $N2 "${N1_ID}" $N1_BAD_ADDR 2> /dev/null)
```

Comment:
What if liberum_cli do not return anything?

Reply:
Well, the test won't pass. Unless you mean the liberum_cli goes dead-locked and hangs. Then the test still won't pass :). Implementing timeouts here is also a subject for improvements in the way all the tests are performed and using a testing framework instead of writing it ourselves from scratch should be discussed